main.tf

```diff
@@ -0,0 +1,71 @@
+########################################################
+### ELB
+########################################################
+resource "aws_elb" "main" {
+  name                        = var.name
+  availability_zones          = var.subnets != null ? null : var.availability_zones
+  name_prefix                 = var.name_prefix
+  security_groups             = var.security_groups
+  subnets                     = var.subnets
+  instances                   = var.instances
+  internal                    = var.internal
+  cross_zone_load_balancing   = var.cross_zone_load_balancing
+  idle_timeout                = var.idle_timeout
+  connection_draining         = var.connection_draining
+  connection_draining_timeout = var.connection_draining_timeout
+  desync_mitigation_mode      = var.desync_mitigation_mode
+  dynamic "listener" {
+    for_each = length(keys(var.listener)) == 0 ? [] : [var.listener]
+    content {
+      instance_port      = lookup(listener.value, "instance_port", null)
```

Comment:
1. For the required  arguments you might want to remove the null option  because it does not apply.
2. Using this block are we able to deploy more than one listener?

Reply:
Yes, with this block one can deploy more than one listener